### PR TITLE
dev-util/clion: bump to 2022.3.

### PR DESCRIPTION
**Pull request by stwnt**
Signed-off-by: Steffen Winter steffen.winter@proton.me

### CODE REVIEW

File: dev-util/clion/clion-2022.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+DESCRIPTION="A complete toolset for C and C++ development"
+HOMEPAGE="https://www.jetbrains.com/clion"
+SRC_URI="https://download.jetbrains.com/cpp/CLion-${PV}.tar.gz"
+
+LICENSE="|| ( IDEA IDEA_Academic IDEA_Classroom IDEA_OpenSource IDEA_Personal )
+	Apache-1.1 Apache-2.0 BSD BSD-2 CC0-1.0 CDDL-1.1 CPL-0.5 CPL-1.0
+	EPL-1.0 EPL-2.0 GPL-2 GPL-2-with-classpath-exception GPL-3 ISC JDOM
+	LGPL-2.1+ LGPL-3 MIT MPL-1.0 MPL-1.1 OFL public-domain PSF-2 UoI-NCSA ZLIB"
+SLOT="0"
+KEYWORDS="~amd64"
```
Comment (ceamac) [inline dev-util/clion/clion-2022.3.ebuild:17]: You removed `~x86`.  Was it dropped by upstream?

File: dev-util/clion/clion-2022.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+# RDEPENDS may cause false positives in repoman.
+# clion requires cmake and gdb at runtime to build and debug C/C++ projects
+RDEPEND="
+	>=app-accessibility/at-spi2-core-2.46.0:2
+	dev-libs/expat
+	dev-libs/glib:2
+	dev-libs/nspr
+	dev-libs/nss
+	dev-util/cmake
```
Comment (ceamac) [inline dev-util/clion/clion-2022.3.ebuild:30]: +`dev-libs/wayland`

File: dev-util/clion/clion-2022.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,105 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit desktop wrapper
+
+DESCRIPTION="A complete toolset for C and C++ development"
+HOMEPAGE="https://www.jetbrains.com/clion"
```
Comment (ceamac) [inline dev-util/clion/clion-2022.3.ebuild:9]:
```suggestion
HOMEPAGE="https://www.jetbrains.com/clion/"
```

**Review by ceamac — CHANGES_REQUESTED** [review]

File: dev-util/clion/clion-2022.3.ebuild (hunk 1 above)
Comment (stwnt) [reply dev-util/clion/clion-2022.3.ebuild:17]: I'm not a sure. I checked the folders in the removed section of the ebuild and all the x86 folders where gone. I have no way of testing it so I removed it.

**Comment by stwnt** [comment]
Ok thats not supposed to happen. I'll open a new PR.

File: dev-util/clion/clion-2022.3.ebuild (hunk 2 above)
Comment (ceamac) [reply dev-util/clion/clion-2022.3.ebuild:31]: You haven't added `dev-libs/wayland` to `RDEPEND`
```
$ lddtree opt/clion-2022.3/jbr/lib/libGLESv2.so
libGLESv2.so => opt/clion-2022.3/jbr/lib/libGLESv2.so (interpreter => none)
    libdl.so.2 => /lib64/libdl.so.2
    libpthread.so.0 => /lib64/libpthread.so.0
    libwayland-client.so.0 => /usr/lib64/libwayland-client.so.0
```

Comment (stwnt) [reply dev-util/clion/clion-2022.3.ebuild:31]: This commit is outdated.

**Comment by ceamac** [comment]
I see you added ~x86 back.  I checked, it doesn't work on x86, so please drop the keyword.

And please fix the commit message (it doesn't contain a summary and the Signed-off-by line is wrong).

Other than that, it looks good.  Thank you!

**Comment by stwnt** [comment]
If you can tell me how. I tried using a colon I the commit message and then git says: aborting commit due to empty commit message

**Comment by ceamac** [comment]
Normally the commit message should look something like this:
```
dev-util/clion: add 2022.3

Signed-off-by: Steffen Winter <steffen.winter@proton.me>
```

But it's easier and safer if you install `dev-util/pkgdev` and commit with `pkgdev commit --signoff`, it will create the commit message for you and will also do some checks before comitting.

Here is some documentation: https://wiki.gentoo.org/wiki/Pkgdev

**Comment by ceamac** [comment]
Your commit message still doesn't have a valid GCO.  It says `Signed-off-by: Steffen Winter steffen.winter@proton.me`.  It should say `Signed-off-by: Steffen Winter <steffen.winter@proton.me>`.  Notice the e-mail address is not between angle brackets.

**Comment by ceamac** [comment]
Also I masked x86, see https://bugs.gentoo.org/887831

**Pull request closed without merging**